src/Microsoft.AspNet.Hosting/WebApplicationBuilder.cs

```diff
@@ -180,7 +113,13 @@ public WebHostBuilder UseServerFactory(IServerFactory factory)
             return this;
         }
 
-        public WebHostBuilder UseStartup(string startupAssemblyName)
+        public WebApplicationBuilder UseLoggerFactory(Action<ILoggerFactory> loggerFactory)
```

Comment:
Sounds more like `ConfigureLoggerFactory` here? Since the user specify what to do with the given `_loggerFactory` instead of providing one.